src/Servers/IIS/IIS/src/Core/IISHttpContext.IO.cs

```diff
@@ -169,6 +169,11 @@ private async Task WriteBody(bool flush = false)
                         // if request is done no need to flush, http.sys would do it for us
                         if (result.IsCompleted)
```

Comment:
The reader is completed after the request pipeline exits: https://github.com/dotnet/aspnetcore/blob/master/src/Servers/IIS/IIS/src/Core/IISHttpContextOfT.cs#L93. It can also be completed due to client disconnect or Abort.

CompleteAsync currently doesn't make it complete early: https://github.com/dotnet/aspnetcore/blob/master/src/Servers/IIS/IIS/src/Core/IISHttpContext.FeatureCollection.cs